layers/+lang/ruby/config.el

```diff
@@ -14,6 +14,9 @@
 (spacemacs|define-jump-handlers enh-ruby-mode)
 (spacemacs|define-jump-handlers ruby-mode)
 
+(defvar ruby-backend nil
+  "The ruby backend for IDE features, possible values are `robe' and `lsp'")
```

Comment:
I'm not sure about having robe by default, while it's a powerful tool, it annoys me how it hijacks the jump to definition and asks to open a ruby shell, which takes a long time to start up after which it can cause the editor to freeze, or it required a `bundle install` and a configured rails environment to give pretty much the same quality of jump to definition that dumb jump gives. Also working on multiple projects with it is painful since you can only have one shell open at a time.

On my work setup I've excluded robe from the package list and never actually felt the need to use it. Before this PR I was considering making a change to put it under a feature flag anyway.

Reply:
Seems like a judgment call to me. I am okay with whatever way you decide to do it but just wanted to give a different perspective.
 
A user who is currently using the vanilla ruby layer would have their autocomplete etc disabled when this change went through as the backed would be `nil` as they are essentially using the same code paths as `robe` atm. Changing default behavior is fine for alignment with conventions but it seems weird to change the default when an opt-in strategy would suffice. ie "You know you don't want it so you turn it off" as opposed to someone who doesn't know a variable exists and has to go find/configure it. 

I like the idea of 3 backends with default being essentially no backend, but I don't know how it fits in the overall scheme of things with usability. Should every language have an autocomplete backend enabled by default or should it be required to turn it on/off per layer? I am sure you have more insight into the layer alignment efforts than I do. So it seems reasonable to me for you to make the choice in how to proceed.